app/components/ConfirmTransaction/index.js

```diff
@@ -0,0 +1,111 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import HdKey from 'ethereumjs-wallet/hdkey'
+import Bip39 from 'bip39'
+import CryptoJs from 'crypto-js'
+
+import 'components/App/styles.css'
+import { assignTransaction } from '../../actions/transactions'
+
+class ConfirmTransaction extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      password: '',
+      errorMessage: '',
+      transaction: {},
+    }
+  }
+
+  componentDidMount = () => {
+    chrome.windows.getCurrent((window) => {
+      this.showTransaction(window.id)
+    })
+  }
+
+  showTransaction = (windowId) => {
+    const { transactions } = this.props
+
+    if (transactions && transactions.length > 0) {
```

Comment:
I think it is clearer if you isolate the code for checking conditions. Something like:
```
if (!transactions || transactions.length === 0) {
  return
}
```

In that way the core code is not spaced and easier to read and to understand.